intro_calculating_cred_grain.md

```diff
@@ -0,0 +1,162 @@
+
+#### Incorporate edge weights
+
+We can adapt the algorithm to account for edge weights on the input graph fairly intuitively – we could add weights to the links by increasing the number of edges for some types of actions (e.g. 👍=1 edge, ♥️=2 edges). In the probability matrix formulation, it’s simple to increase the transition probability of one edge. Then, instead of assigning equal probability to all outbound edges, we can simply normalize the weights so that they sum to one and this provides us with transition probabilities needed for <img src="https://render.githubusercontent.com/render/math?math=P_{uv}">.
+
+#### Incorporate node weights
+
+There’s also a cleverly simple way to account for the existence of node weights. One requirement for the existence of a stationary distribution that was skipped is the concept of ergodicity, wherein every state must be reachable from any other state, given a long enough probabilistic walk across the graph’s edges†. To achieve ergodicity, SourceCred uses a "teleportation" system by adding a probability, ⍺, of teleporting to a random location in the graph. SourceCred currently implements this with a seed node, which all nodes teleporting to with likelihood ⍺. From the seed node, the probability of landing on a given node in the graph is then given by the node weights. If we use _A_ to describe the weights of each node in a vector form, then our old transition matrix <img src="https://render.githubusercontent.com/render/math?math=P_{uv}"> becomes the new transition matrix <img src="https://render.githubusercontent.com/render/math?math=(1-\alpha)P_{uv} + \alpha A">. The stationary distribution is then the vector <img src="https://render.githubusercontent.com/render/math?math=\vec v"> such that <img src="https://render.githubusercontent.com/render/math?math=\left((1-\alpha)P_{uv} + \alpha A\right)\vec v = \vec v">.
+
+#### The origin of Cred
+
+In this way, Cred originates at the weighted nodes and flows out to actions from there. Interestingly, this is mathematically equivalent to adding edges to each and every node, with edge-weight proportional to ⍺ times the node weight. We can interpret the necessity of teleportation in the following way: because some contributions may receive but flow no cred, there need to be sources of Cred to calculate an equilibrium value distribution, which requires some contributions to mint cred.
+
+#### Cred-minting philosophy
```

Comment:
I'd like to see an explanation of cred minting earlier in the doc. I've been explaining it like:

- Cred flows on the graph, but it needs to "start" somewhere
- We have it start on nodes that have non-zero weight. Thus, we can say that these nodes "mint" new cred.
- The total amount of cred received by contributors will equal the total node weight, i.e. the total amount minted.
- As an example, if I boost a pull request so that it's node weight goes from 0 to 10, then there is 10 new cred in existence, and it will flow out from that pull request to everything that touches the pull.

Reply:
I moved this up to the beginning of the CredRank section, and included your points in a slightly summarized way.  Open to comments on the new version, which should be much closer to this.